qiime2/core/archive/tests/test_provenance.py

```diff
@@ -262,6 +262,20 @@ def test_ts_to_date(self, mocked_tzlocal):
         self.assertEqual(obs, exp)
         self.assertTrue(mocked_tzlocal.called)
 
+    def test_prov_rename(self):
+        viz, = dummy_plugin.actions.no_input_viz()
+
+        viz_p_dir = viz._archiver.provenance_dir
+        self.assertTrue(viz_p_dir.exists())
+
+    @mock.patch('qiime2.core.path.ProvenancePath.rename',
+                side_effect=FileExistsError)
+    def test_prov_rename_file_exists(self, _):
+        viz, = dummy_plugin.actions.no_input_viz()
+
+        viz_p_dir = viz._archiver.provenance_dir
```

Comment:
I'm not sure this test is enough, as that directory may exist (while being empty). Perhaps we check for the existence of `action.yaml` which should be in there?